Remotion/ObjectBinding/Sample/SessionStateReflectionBusinessObjectStorageProvider.cs

```diff
@@ -0,0 +1,83 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Remotion.Utilities;
+using Remotion.Web.Infrastructure;
+
+namespace Remotion.ObjectBinding.Sample
+{
+  public class SessionStateReflectionBusinessObjectStorageProvider : IReflectionBusinessObjectStorageProvider
+  {
+    private const string c_sessionStateEntryName = "SessionStateReflectionBusinessObjectStorageProviderEntry";
+
+    private readonly IHttpContextProvider _httpContextProvider;
+    private readonly IReflectionBusinessObjectStorageProviderFactory _reflectionBusinessObjectStorageProviderFactory;
+
+    public SessionStateReflectionBusinessObjectStorageProvider (
+        IHttpContextProvider httpContextProvider,
+        IReflectionBusinessObjectStorageProviderFactory reflectionBusinessObjectStorageProviderFactory)
+    {
+      ArgumentUtility.CheckNotNull ("httpContextProvider", httpContextProvider);
+      ArgumentUtility.CheckNotNull ("reflectionBusinessObjectStorageProviderFactory", reflectionBusinessObjectStorageProviderFactory);
+
+      _httpContextProvider = httpContextProvider;
+      _reflectionBusinessObjectStorageProviderFactory = reflectionBusinessObjectStorageProviderFactory;
+    }
+
+    /// <inheritdoc />
+    public IReadOnlyCollection<Guid> GetObjectIDsForType (Type type)
+    {
+      ArgumentUtility.CheckNotNull ("type", type);
+
+      var reflectionBusinessObjectStorageProvider = GetReflectionBusinessObjectStorageProviderForCurrentSession();
+      return reflectionBusinessObjectStorageProvider.GetObjectIDsForType (type);
+    }
+
+    /// <inheritdoc />
+    public Stream GetReadObjectStream (Type type, Guid id)
+    {
+      ArgumentUtility.CheckNotNull ("type", type);
+
+      var reflectionBusinessObjectStorageProvider = GetReflectionBusinessObjectStorageProviderForCurrentSession();
+      return reflectionBusinessObjectStorageProvider.GetReadObjectStream (type, id);
+    }
+
+    /// <inheritdoc />
+    public Stream GetWriteObjectStream (Type type, Guid id)
+    {
+      ArgumentUtility.CheckNotNull ("type", type);
+
+      var reflectionBusinessObjectStorageProvider = GetReflectionBusinessObjectStorageProviderForCurrentSession();
+      return reflectionBusinessObjectStorageProvider.GetWriteObjectStream (type, id);
+    }
+
+    private IReflectionBusinessObjectStorageProvider GetReflectionBusinessObjectStorageProviderForCurrentSession ()
+    {
+      var httpContext = _httpContextProvider.GetCurrentHttpContext();
+      var reflectionBusinessObjectStorageProvider = httpContext.Session[c_sessionStateEntryName] as IReflectionBusinessObjectStorageProvider;
+      if (reflectionBusinessObjectStorageProvider != null)
```

Comment:
invert if, reduces code to one return and also removes the not-complexity from the condition.